API Descriptions Update

Bashamega opened the pull request:
fixes #1937

saschanaz commented:
We still depend on that, so we can't just remove it until we get a new mdn/content dependency?

Bashamega commented:
> We still depend on that, so we can't just remove it until we get a new mdn/content dependency?

I know

saschanaz commented on src/build.ts:
```diff
@@ -85,6 +86,101 @@ async function emitFlavor(
+
+  const OWNER = "mdn";
+  const REPO = "content";
+  const BASE_PATH = "files/en-us/web/api";
```
I think maybe we should add a submodule so that we can make the build more reproducible locally

Bashamega replied:
Sure, I can do that
I will get started on that

Bashamega commented:
Hello @saschanaz,
I have created a script to generate the JSON, ran the baseline command, and fixed some linting problems, but I don't know how to solve the error you see in the logs. It is just one error.

saschanaz commented:
Maybe `npm i`?

Bashamega commented:
> Maybe `npm i`?

Hello:)
How will `npm i` help?
I ran it and it didn't do anything

saschanaz commented:
I mean `npm i && npm run build && npm run baseline-accept`? Because it seem like it's caused by mismatch of installed dependency version.

Bashamega commented:
> I mean `npm i && npm run build && npm run baseline-accept`? Because it seems like it's caused by mismatch of dependency version. `npm i` fixes that.
oh okay

saschanaz commented on baselines/audioworklet.generated.d.ts:
```diff
@@ -289,7 +297,11 @@ declare var ByteLengthQueuingStrategy: {
 
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CompressionStream) */
+/**
+ * Of the {{domxref('Compression Streams API','','',' ')}} is an API for compressing a stream of data.
```
This looks weird. Now that we are at it, maybe we should remove:

https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/cede2ca8e8dbf8344de3f5fc341b01e7ff6c9458/src/build.ts#L94-L113

Because I don't think it makes sense to deal with all kinds of human language without accidentally breaking things.

Bashamega replied:
I think it would be easier if we converted the Markdown to HTML. I can utilize Cheerio like the mdn and get the summary like the mdn. If you think this is a good way to do it, I can implement it.

saschanaz replied:
Not sure why that's needed here, can you explain more?

saschanaz replied:
Oh you mean the domxref thing. Hmm.

Bashamega replied:
Because the reason that I made all these regexes is to avoid the weird template code. But if I convert the markdown to HTML first, I think it would be easier to extract the data without the weird templates

Bashamega replied:
Oh

saschanaz replied:
If using Cheerio is easy to remove domxref thing then I guess it's worth trying!

Bashamega replied:
> If using Cheerio is easy to remove domxref thing then I guess it's worth trying!

Okay

saschanaz commented on package.json:
```diff
@@ -13,6 +13,7 @@
   ],
   "scripts": {
     "build": "tsc && node ./lib/build.js",
+    "generate": "tsc && node ./lib/generateDescriptions.js",
```
Can this be part of `build` instead (ideally called by lib/build.js?)

Bashamega replied:
Sure, I can do that :)

Bashamega commented:
We need to figure out a way to compile the markdown without the weird templates.

saschanaz commented on baselines/audioworklet.generated.d.ts:
```diff
@@ -999,7 +1059,7 @@ declare var TransformStreamDefaultController: {
 
 /**
- * The URL interface represents an object providing static methods used for creating object URLs.
+ * The **URL** interface is used to parse, construct, normalize, and encode . It works by providing properties which allow you to easily read and modify the components of a URL.
```
Hmm, encode what? It seems links are gone.

Bashamega replied:
I will try to fix it

saschanaz reviewed (commented):
Wow, getting a lot better!

Bashamega commented:
> Wow, getting a lot better!

Thank you

Bashamega commented:
Hello @saschanaz 
I have tried to run `npm i && npm run build && npm run baseline-accept` but nothing got updated.
How can i fix the failing test?

Bashamega commented:
Hello @saschanaz 
I have fixed the builds.
Please review :)

Bashamega commented:
Hello @saschanaz,

I’ve completed the necessary changes in this PR, and I believe it’s ready for review and merging. Could you take a look when you have a moment?

Thanks!

saschanaz commented on baselines/audioworklet.generated.d.ts:
```diff
@@ -339,7 +355,7 @@ declare var CustomEvent: {
 
 /**
- * An abnormal event (called an exception) which occurs as a result of calling a method or accessing a property of a web API.
+ * The **DOMException** interface represents an abnormal event (called an **exception**) that occurs as a result of calling a method or accessing a property of a web API. This is how error conditions are described in web APIs.
```
This what? 🤔 Should we try to get only the first sentence?

Bashamega replied:
Sure, I can do that. Do you have any other changes?

Bashamega replied:
Done

saschanaz commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,114 @@
+    .replace(
+      /\{\{\s*(Glossary|HTMLElement|SVGAttr|SVGElement|cssxref|jsxref|HTTPHeader)\s*\(\s*["']((?:\\.|[^"\\])*?)["'].*?\)\s*\}\}/gi,
+      "$2",
+    ) // Extract first argument from multiple templates, handling escaped quotes & spaces
```
I guess the plan to use the parser didn't work well? 👀 (For now this is fine as-is in your patch)

Bashamega replied:
Yes, it didn't work out

saschanaz replied:
Okay. I think given we have this function here we can probably remove removeVerboseIntroductions() from src/build.

saschanaz commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,114 @@
+    const folders = await getFolders(basePath);
+    if (folders.length) {
+      const data = await getIndexMdContents(folders);
+      await fs.writeFile(outputFile, JSON.stringify(data, null, 2), "utf-8");
```
We don't have to store a file, emitDom should instead receive an object.

Bashamega replied:
This would cause the test workflow to break because, by default, it won't clone the submodule. As a result, it will return an empty array, and during the build process, it will report that the baselines are not the same. You can modify the workflow to clone the submodule, but unfortunately, I can't do that since I don't have permission to update the workflows.

Bashamega replied:
> > I can't do that since I don't have permission to update the workflows.
> 
> The CI won't run but we can still verify it locally and eventually merge it, right?

Yes

saschanaz commented on .gitignore:
```diff
@@ -285,4 +285,5 @@ inputfiles/browser.webidl.json
-deploy/generated
\ No newline at end of file
+deploy/generated
+.early.coverage
```
What is this file?

Bashamega replied:
Sorry:)
It is something locally for me, I will remove it

saschanaz commented on .gitmodules:
```diff
@@ -0,0 +1,3 @@
+[submodule "mdn-content"]
+	path = mdn-content
```
Can this go inside `inputfiles/`?

Bashamega replied:
Sure

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -6918,7 +7175,7 @@ declare var DOMMatrixReadOnly: {
 
 /**
- * Provides the ability to parse XML or HTML source code from a string into a DOM Document.
+ * The **DOMParser** interface provides
```
Err, looks like some files have newlines within a sentence: https://github.com/mdn/content/blob/a52689c74c6c89f45c54447bb148e54ed320db62/files/en-us/web/api/domparser/index.md?plain=1#L3

Bashamega commented:
**Hello @saschanaz,**  

I've made the following updates:  
- Removed an extra folder that was mistakenly added to `.gitignore`.  
- Updated the directory path from `/mdn-content` to `/inputfiles/mdn`.  
- Removed the cleaner function from the generator file since it's already handled in the build file.  
- Changed the splitting logic to split by sentence instead of by new line.    

Let me know if any further changes are needed!

Bashamega commented:
> Hmm. What happens if you modify workflow file here? Does it outright reject your commit? It shouldn't as it's still your own fork 🤔

I have tried it again, and it worked. I guess yesterday it didn't work because I had an issue with my laptop and had to use Gitpod.

saschanaz commented on .github/workflows/test_typescript.yml:
```diff
@@ -14,6 +14,13 @@ jobs:
+
+      - name: Initialize and update submodules
+        run: |
+          git submodule update --init --recursive  # Explicitly initialize submodules
```
This is test_typescript, you should also update ci.yml.

saschanaz replied:
And also deploy.yml.

Bashamega commented:
Hello @saschanaz 
I have fixed the templates. Please review 😃

saschanaz commented on src/build.ts:
```diff
@@ -394,5 +418,4 @@ async function emitDom() {
     }
   }
 }
-
```
Maybe restore this?

saschanaz replied:
(Still a removed newline)

saschanaz commented on src/build.ts:
```diff
@@ -195,7 +196,30 @@ async function emitDom() {
+      .replace(/\\(["'])/g, "$1") // Remove backslashes from escaped quotes
+      .replace(/&lt;/g, "<")
+      .replace(/&gt;/g, ">") // Decode HTML entities
+      .replace(/`([^`]+)`/g, "$1") // Keep inline code readable
```
Why do we remove backticks? 👀

Bashamega replied:
Done

saschanaz replied:
Hmm? It still removes backticks 👀

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -3919,7 +4018,11 @@ declare var CSSMathClamp: {
 
-/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/CSSMathInvert) */
+/**
+ * The **CSSMathInvert** interface of the CSS Typed Object Model API represents a CSS calc used as calc(1 / ). It inherits properties and methods from its parent CSSNumericValue.
```
This lacks `<value>` in `calc(1 / )`, that's tricky... maybe we should have a real markdown parser for that. I think that can be done in a followup and not necessarily here.

Bashamega replied:
I have fixed it

saschanaz replied:
We still lack backticks here, and in some markdown implementations lacking backticks may mean `<value>` won't be visible at all. (e.g. if it uses GFM)

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -8348,7 +8631,7 @@ interface ElementEventMap {
 
 /**
- * Element is the most general base class from which all objects in a Document inherit. It only has methods and properties common to all kinds of elements. More specific classes inherit from Element.
+ * **Element** is the most general base class from which all element objects (i.e.
```
Oh no. This is tricky too without having an actual language processor library. Something for a followup.

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -9511,7 +9859,7 @@ declare var GamepadButton: {
 
 /**
- * This Gamepad API interface contains references to gamepads connected to the system, which is what the gamepad events Window.gamepadconnected and Window.gamepaddisconnected are fired in response to.
+ * This Gamepad API interface contains references to gamepads connected to the system, which is what the gamepad events {{domxref("Window.gamepadconnected_event", "gamepadconnected")}} and {{domxref("Window.gamepaddisconnected_event", "gamepaddisconnected")}} are fired in response to.
```
Still some domxref? 👀

Bashamega replied:
I have fixed it. I forgot to remove templates when I have migrated it to the build file

Bashamega commented:
I will try to fix more

saschanaz commented on .github/workflows/ci.yml:
```diff
@@ -7,6 +7,13 @@ jobs:
+
+      - name: Initialize and update submodules
+        run: |
+          git submodule update --init --recursive  # Explicitly initialize submodules
```
This shouldn't be needed anymore, right?

Bashamega replied:
I have forgot to remove them. I have removed them

Bashamega commented:
Hello @saschanaz 
I forgot to remove the backticks. They are now removed

saschanaz commented on baselines/ts5.5/dom.generated.d.ts:
```diff
@@ -8624,7 +8624,7 @@ interface ElementEventMap {
 
 /**
- * **Element** is the most general base class from which all element objects (i.e.
+ * **`Element`** is the most general base class from which all element objects (i.e.
```
(Something for a followup)

Bashamega replied:
This is because we do not check for abbreviations

Bashamega replied:
I have fixed it

Bashamega replied:
Hello @saschanaz
[This pr](https://github.com/mdn/content/pull/38905) should fix this problem. Once it is merged I will regenerate the baselines

saschanaz commented on baselines/webworker.generated.d.ts:
```diff
@@ -2623,11 +2774,7 @@ interface EXT_color_buffer_half_float {
- *
- * [MDN Reference](https://developer.mozilla.org/docs/Web/API/EXT_frag_depth)
- */
+/** [MDN Reference](https://developer.mozilla.org/docs/Web/API/EXT_frag_depth) */
```
Hmm? How did we lose it 🤔

Bashamega replied:
I have fixed it

saschanaz commented on baselines/serviceworker.generated.d.ts:
```diff
@@ -8477,7 +8744,12 @@ declare namespace WebAssembly {
+/** The **`console`** object provides access to the debugging console (e.g., the Web console in Firefox). */
+/**
+ * The **`console`** object provides access to the debugging console (e.g., the Web console in Firefox).
+ *
```
This gets double comments? 🤔

saschanaz reviewed (commented):
Skimmed again and mostly looks nice, except a few now-missing comments and duplicated comments. This is HUGE, thank you for the effort here!

saschanaz commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,91 @@
+        !line.startsWith(">") &&
+        !line.startsWith("{{"),
+    )
+    .join(" ");
```
transformVerbosity should be merged here. (And it's not about "verbosity" anymore)

Bashamega replied:
I don't understand

saschanaz commented on baselines/dom.generated.d.ts:
```diff
@@ -8348,7 +8675,7 @@ interface ElementEventMap {
 
 /**
- * Element is the most general base class from which all objects in a Document inherit. It only has methods and properties common to all kinds of elements. More specific classes inherit from Element.
+ * **`Element`** is the most general base class from which all element objects (i.e.
```
(This is still a problem)

Bashamega replied:
Okay

saschanaz commented on src/build.ts:
```diff
@@ -160,7 +139,7 @@ async function emitDom() {
     for (const [key, value] of Object.entries(descriptions)) {
       const target = idl.interfaces!.interface[key] || namespaces[key];
       if (target && !value.startsWith("REDIRECT")) {
```
REDIRECT seems like an old thing that can be removed now.

saschanaz commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,91 @@
+
+  // Extract the first sentence (ending in . ! or ?)
+  const sentenceMatch = normalizedText.match(
+    /(.*?(?<!\.\s(?:i\.e|e\.g))([.!?]))(?:\s|$)/i,
```
This doesn't really solve the "e.g." case, as replacing this with `/(.*?[.!?])(?=\s|$)/` doesn't change anything. I'm okay with deferring that issue for now, but in that case we should revert this regex, to prevent giving false sense that this solves the case when it does not.

Bashamega commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,91 @@
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
```
Done

Bashamega commented on src/generateDescriptions.ts:
```diff
@@ -0,0 +1,91 @@
+import { promises as fs } from "fs";
```
Done

saschanaz commented on package.json:
```diff
@@ -58,10 +58,12 @@
     "globals": "^15.9.0",
     "jsonc-parser": "^3.2.1",
     "node-fetch": "^3.3.2",
+    "path": "^0.12.7",
```
This and `url` below too.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -1,9 +1,10 @@
+const basePath = new URL(
+  "../../inputfiles/mdn/files/en-us/web/api/",
+  import.meta.url,
+).pathname;
```
Using pathname still requires using `path` elsewhere, but the file system API can understand URL directly.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -0,0 +1,85 @@
+    const entries = await fs.readdir(dirPath, { withFileTypes: true });
+    return entries
+      .filter((entry) => entry.isDirectory())
+      .map((entry) => path.join(dirPath, entry.name));
```
Here for example, this could be `new URL(entry.name, dirPath)` and return `Promise<URL[]>`.

saschanaz commented on src/build.ts:
```diff
@@ -178,24 +157,32 @@ async function emitDom() {
-    }
-
-    return description;
+  function transformVerbosity(description: string): string {
```
We should move all MDN comment generation code to `mdn-comments.ts`. IMO this can be merged right into `extractSummary()`.

Bashamega commented:
I have fixed all the issues @saschanaz

saschanaz commented on .github/workflows/update-core-deps.yml:
```diff
@@ -12,6 +12,17 @@ jobs:
+        run: |
+          git submodule update --init
+          cd inputfiles/mdn  # Replace with actual submodule path
+          git checkout main  # Change if using a different branch
```
We are certainly using a different branch (well, no branch actually), so no "if". And `git switch` is preferred for branches nowadays.

saschanaz commented on .github/workflows/update-core-deps.yml:
```diff
@@ -1,4 +1,4 @@
-name: Update core dependencies
+name: Update core dependencies and submodules
```
The MDN submodule is also a core dependency so we can skip it. Same for the changes below.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -32,34 +50,37 @@ function extractSummary(markdown: string): string {
+async function getFolders(dirPath: URL): Promise<URL[]> {
   try {
-    const entries = await fs.readdir(dirPath, { withFileTypes: true });
+    const entries = await fs.readdir(fileURLToPath(dirPath), {
```
We don't need `fileURLToPath`, you can pass `dirPath` directly.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -32,34 +50,37 @@ function extractSummary(markdown: string): string {
 
     try {
-      const content = await fs.readFile(indexPath, "utf-8");
+      const content = await fs.readFile(fileURLToPath(indexPath), "utf-8");
```
Same here, `fileURLToPath` can be skipped.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -32,34 +50,37 @@ function extractSummary(markdown: string): string {
 
       // Improved title extraction
       const titleMatch = content.match(/title:\s*["']?([^"'\n]+)["']?/);
+      const filename = fileURLToPath(folder).split("/").pop();
```
You can do `path.basename(folder.toString())`

Bashamega commented:
Done @saschanaz

saschanaz commented on .github/workflows/update-core-deps.yml:
```diff
@@ -34,13 +41,13 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.TS_GITHUB_BOT_AUTH }}
       - id: git-diff
-        run: git diff --quiet HEAD baselines
+        run: git diff --quiet HEAD baselines submodules
```
There's no path `submodules`, and just comparing `baselines` is enough, nothing more is needed here.

saschanaz commented on .github/workflows/update-core-deps.yml:
```diff
@@ -34,13 +41,13 @@ jobs:
         with:
-          commit-message: "🤖 Update core dependencies"
-          title: "Update core dependencies"
+          commit-message: "🤖 Update core dependencies and submodules"
+          title: "Update core dependencies and submodules"
```
Please also revert these two.

saschanaz commented on src/build/mdn-comments.ts:
```diff
@@ -0,0 +1,106 @@
+  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+}
+
+async function getFolders(dirPath: URL): Promise<URL[]> {
```
getDirectories, to match other names (it's readdir and dirPath, not readfolder and folderPath)

```suggestion
async function getDirectories(dirPath: URL): Promise<URL[]> {
```

saschanaz reviewed (commented):
The only blocking change is the workflow thing, otherwise this looks mostly fine. 🎉

Bashamega commented:
Done @saschanaz

saschanaz commented on README.md:
```diff
@@ -88,7 +88,7 @@ If you are familiar with Web IDL, you may also want to check whether the upstrea
 - `removedTypes.jsonc`: types that are defined in the spec but should be removed.
 - `comments.json`: comment strings to be embedded in the generated .js files.
-- `deprecatedMessage.json`: the reason why one type is deprecated. The reason why it is a separate file rather than merge in comment.json is mdn/apiDescriptions.json would also possibly be deprecated.
+
```
Removing that is the last request from me

Bashamega commented:
Done @saschanaz

Bashamega commented:
Hello @saschanaz,

My PR to the MDN repo has been merged, and I’ve synced the repo to resolve the abbreviation issue.

saschanaz commented:
LGTM, thanks! (This may also require Microsoft review, but let me try given there's no maintainers tag...)

Bashamega commented:
Hello @sandersn
Can you please check this PR and merge it?
Thank you
CC - @saschanaz

sandersn commented on .github/workflows/update-core-deps.yml:
```diff
@@ -12,6 +12,13 @@ jobs:
+          submodules: true  # Ensure submodules are checked out
+
+      - name: Update submodules
+        working-directory: inputfiles/mdn  # Replace with actual submodule path
```
these comments look left over from an example, since `inputfiles/mdn` is the actual submodule path. Can you remove them?

sandersn reviewed (changes requested):
One tiny cleanup request. 

I'm not a fan of submodules but if @saschanaz is OK with the overhead of working with them it's fine with me.

saschanaz commented:
I considered adding it as a npm package but that causes a terribly deep dependency tree, so I think submodule is the best choice for now.

Bashamega commented:
Thank you @sandersn for your review. I have updated the workflow

Bashamega commented:
> One tiny cleanup request. 
> 
> I'm not a fan of submodules but if @saschanaz is OK with the overhead of working with them it's fine with me.

Hello @sandersn, I understand you worry that it won't be updated regularly because the update dependencies workflow will auto update it. So I don't think you should worry about it 🙂

Bashamega commented:
Hello @jakebailey, and @sandersn 
I have applied all the requested changes. Can we merge this?

jakebailey commented:
I guess I'm of the opinion that adding a submodule is a bad idea 😄 It's the entire MDN repo! Is there no other way?

> I understand you worry that it won't be updated regularly because the update dependencies workflow will auto update it. So I don't think you should worry about it 

The updating script now just keeps pulling main from the submodule, which I guess works but sounds like it's going to be a lot of churn, since before the dep update may not actually do anything, but of course MDN is going to be updated frequently...

saschanaz commented:
1. Manually download the tree, as initially this PR was doing. You'll have to download the same size for every update (even a small update would trigger a full download), and also have to manually maintain the code to make sure the local download matches the in-tree reference, caching if one has to switch to different branches, etc etc.
2. `git subtree --squash`. This would allow downloading only some part of the repo. But the same issue exists: even a small update would trigger a full download.
3. `git subtree` without `--squash`. I don't think we want to merge all the commits to this repository. Plain No.

jakebailey commented:
I'm a big fan of subtrees but they're definitely the wrong tool for this particular job (pulling in someone else's code but not modifying it or pushing back) 😄

Maybe a submodule is fine. I haven't looked deeply to see what all is done here, so this is mainly a gut reaction. Will harder look soon.

Bashamega commented:
Hello @jakebailey,
Technically, there is another option: scraping the MDN docs programmatically. However, this would require more code and may not be as efficient. I believe using a submodule is the best approach, but if you have another suggestion, I'm open to ideas!

jakebailey commented:
Read the PR, and yeah, I think it's untenable to do anything other than something that produces a local clone. I was thinking "what about asking github for the raw files", but this quite literally loops over every single markdown file in MDN (more or less), so that won't work.

jakebailey commented on package-lock.json:
Why is this file changing? `package.json` hasn't changed, so I wouldn't expect a lockfile update here.

jakebailey commented on README.md:
```diff
@@ -88,7 +88,6 @@ If you are familiar with Web IDL, you may also want to check whether the upstrea
 - `overridingTypes.jsonc`: types that are defined in the spec but have TypeScript-friendly modifications in the json files.
 - `removedTypes.jsonc`: types that are defined in the spec but should be removed.
 - `comments.json`: comment strings to be embedded in the generated .js files.
-- `deprecatedMessage.json`: the reason why one type is deprecated. The reason why it is a separate file rather than merge in comment.json is mdn/apiDescriptions.json would also possibly be deprecated.
```
This line is being removed, but the `deprecatedMessage.json` file still exists in the repo. Should it be removed?

Bashamega replied:
This was a mistake

jakebailey replied:
Also, given we're going to depend on mdn, I do wonder if we can extract this information from there too. It seems like mdn includes warnings for deprecations.

Bashamega replied:
Yeah, but this PR is already huge, I think making it a separate PR would be better. I am planning on making more PRs like for more messages

jakebailey replied:
Yes, just thinking aloud.

saschanaz replied:
That info is already being imported with more readable form via `@mdn/browser-compat-data`.

jakebailey commented on package-lock.json:
```diff
@@ -348,9 +348,9 @@
-      "version": "6.0.4",
-      "resolved": "https://registry.npmjs.org/@mdn/browser-compat-data/-/browser-compat-data-6.0.4.tgz",
-      "integrity": "sha512-d9Em7f44DDP1BkJdqBvbjohC25efLgdGoNJvobxm5iMN0EGK677pC83SBjyPmU7OObc9s8Xu+Sv00dzKS+Xdng==",
+      "version": "6.0.5",
```
This is still modified even though nothing changed in the repo that would do this.

Consider doing something like `git checkout main -- package-lock.json` or something.

jakebailey commented on .github/workflows/update-core-deps.yml:
```diff
@@ -12,6 +12,13 @@ jobs:
+          submodules: true  # Ensure submodules are checked out
+
+      - name: Update submodules
+        working-directory: inputfiles/mdn
+        run: |
+          git pull origin main
```
I would think a better way to do this would be `git submodule update --init --remote`? That way, the gitmodules config the source of truth for pulling new refs, without hardcoding any dirs or branch names.

Bashamega replied:
done

jakebailey commented on src/build/mdn-comments.ts:
```diff
@@ -0,0 +1,106 @@
+}
+
+export async function generateDescription(): Promise<Record<string, string>> {
+  try {
```
In this function I would do a `stat` to verify that `basePath` exists. If it doesn't, we should print out a message that says something like:

> MDN submodule does not exist; try running `git submodule update --init`

Or something.

Honestly, I feel like printing errors here in this function is wrong; we should just throw and let the script break, since something is very off.

Bashamega replied:
done

jakebailey reviewed (approved):
Theoretically seems good to me; @saschanaz seem alright as a first go?

saschanaz commented on .github/workflows/update-core-deps.yml:
```diff
@@ -12,6 +12,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          submodules: true  # Ensure submodules are checked out
```
But to update it doesn't make much sense to have `depth=1` because it would cause double full downloads?

saschanaz replied:
Maybe we can just skip this line when we call `git submodule update` below anyway.

jakebailey replied:
No, there's no inefficiency; git will correctly handle reusing data and not double download. Mainly I was just thinking about whether or not something like:

```
git clone --depth=1 ...
git pull
```

Works for new refs. Which, I think it does.

saschanaz replied:
Hmm, interesting, I guess it's fine then. 👍🏻

Bashamega replied:
Hello @jakebailey 
According to the documentation for this action, the default is set to one.

jakebailey replied:
Yes, that's what I'm saying it does. I haven't personally attempted to update a submodule that was cloned with depth 1, but I will probably go try that quick.

Bashamega commented:
Hi @saschanaz, @jakebailey, @sandersn 👋

I've looked into ways to clone only specific parts of a repo without fetching the entire repository. Git doesn't support this natively, but I found a practical workaround using [SVN (Subversion)](https://subversion.apache.org/).

Although it's an older tool, it’s still supported by GitHub and allows downloading a single directory without pulling the whole repo. This might be a viable solution for our case—just wanted to share as a proposal for consideration.

jakebailey commented:
I don't think we need to do anything fancy. The submodule is fine.

Bashamega commented:
> I don't think we need to do anything fancy. The submodule is fine.

Okay, do you have any estimation of when this pr will be merged?

saschanaz commented:
SVN is *not* supported by GitHub https://github.blog/news-insights/product-news/sunsetting-subversion-support/ (I'm pretty surprised that it was supported at all)

saschanaz commented:
I see no blocker to not merge.

Bashamega commented:
> SVN is _not_ supported by GitHub https://github.blog/news-insights/product-news/sunsetting-subversion-support/ (I'm pretty surprised that it was supported at all)

Good to know, I didn't know

saschanaz commented:
I mean only if I could... 🤷🏻‍♀️

LGTM

Bashamega commented:
Hello @jakebailey 
Sorry for pinging you, but I think this PR is ready to be merged

jakebailey commented:
I'll merge it now, I had just replied late at night 😄

Bashamega commented:
> I'll merge it now, I had just replied late at night 😄

Thank you very much for merging this huge PR. You just made my day :)
